### PR TITLE
UIROLES-43 Use ChunkedCQL for request

### PR DESCRIPTION
**Pull request by ryandberger**
- Relies on https://github.com/folio-org/stripes-core/pull/1435 being merged first.
- Use ChunkedCQL to handle requests which may cause max URL length to be exceeded by splitting and batching requests as needed.
- By using this method, `limit=1000` is also passed as a URL param, which is desirable to improve result set (which is a known odd behavior of FOLIO CQL back-end).
- Convert API error case to use `window.alert()` instead of `console.error()`.

### CODE REVIEW

File: src/settings/components/RoleDetails/AssignUsers.js [hunk 1]
```diff
@@ -65,7 +65,7 @@ const AssignUsers = ({ selectedUsers, roleId, refetch }) => {
         }
       }
     })();
-  }, [isSuccess, roleDetails, isAssignUsers]); // eslint-disable-line react-hooks/exhaustive-deps
+  }, [isAssignUsers]); // eslint-disable-line react-hooks/exhaustive-deps
```
Comment (zburke) [inline src/settings/components/RoleDetails/AssignUsers.js:68]: I realize the original `eslint-disable` was not part of this PR, but doubling-down on it without explaining it remains a problem. This rule is pretty smart and gets things right most of the time. If you think you're smarter than lint, please add a comment explaining the circumstances.

File: src/hooks/useRoleByUserIds.js [hunk 2]
```diff
@@ -1,26 +1,23 @@
-import { useNamespace, useOkapiKy } from '@folio/stripes/core';
-import { useQuery } from 'react-query';
+import { useChunkedCQLFetch } from '@folio/stripes/core';
 
 function useRoleByUserIds(users) {
```
Comment (zburke) [inline src/hooks/useRoleByUserIds.js:3]: It took me a while to figure out what this hook does, and what the underlying data structure is (we're querying a join-table here with one-half of the PK (the userId, the other half being the roleId). Documentation would be really helpful to make this a bit less opaque.

**Review by zburke — APPROVED** [review]
The data structure behind `useRoleByUserIds` was hard for me to follow without documentation, in part because the name lead me to think it would return `role` records but it really returns `role-user` records from a join table. 

Please clean up lint errors before you commit.

**Review by aidynoJ — APPROVED** [review]

**Comment by ryandberger** [comment]
> The data structure behind `useRoleByUserIds` was hard for me to follow without documentation, in part because the name lead me to think it would return `role` records but it really returns `role-user` records from a join table.
> 
> Please clean up lint errors before you commit.

I'll rename it to be `useRoleUsersByUserIds` and add documentation. Hopefully that helps distinguish it from `useRoleById`.

File: src/settings/components/RoleDetails/AssignUsers.js (hunk 1 above)
Comment (ryandberger) [reply src/settings/components/RoleDetails/AssignUsers.js:68]: If I add all values (11 in total) it causes an infinite loop. But that probably points to a flaw in this component. It's a little awkward since I am using the callback from the FindUser Plugin to trigger `useEffect()`. May be better to move more of this work into the callback itself. I'll try that and update if successful.

Comment (ryandberger) [reply src/settings/components/RoleDetails/AssignUsers.js]: Was impractical because of how `useChunkedCQLFetch` is called, so instead I made a comment to justify `eslint-disable`.

File: src/hooks/useRoleByUserIds.js (hunk 2 above)
Comment (ryandberger) [reply src/hooks/useRoleByUserIds.js]: Done
